### PR TITLE
NAS-128166 / 24.10 / Add layout selector for new dashboard

### PR DESCRIPTION
**Pull request by denysbutenko**
Go to the new dashboard and press the `Configure` button for testing. You'll see empty template select and new layout selector component.

### CODE REVIEW

File: src/app/pages/dashboard/components/widget-group-form/widget-editor-group/widget-editor-group.component.ts [hunk 1]
```diff
@@ -19,4 +21,12 @@ export class WidgetEditorGroupComponent {
   @Input() group: WidgetGroup;
   @Input() selection = 0;
   @Output() selectionChange = new EventEmitter<number>();
+  protected form = this.formBuilder.group({
+    template: [''],
+    layout: [WidgetGroupLayout.Full],
+  });
+  readonly layoutsMap = widgetGroupIcons;
+  templateOptions$ = of([]);
```
Comment (RehanY147) [inline src/app/pages/dashboard/components/widget-group-form/widget-editor-group/widget-editor-group.component.ts:29]: Let's add a TODO here so it doesn't get overlooked

**Review by RehanY147 — CHANGES_REQUESTED** [review]
What is the scenario for approving the PR? I can see the dotted editable areas. I can click the edit button and open the edit form. but it's missing a Save button or cancel. I think we should add those as well. If not, please add a comment why.

**Comment by denysbutenko** [comment]
> What is the scenario for approving the PR? I can see the dotted editable areas. I can click the edit button and open the edit form. but it's missing a Save button or cancel. I think we should add those as well. If not, please add a comment why.

The scope of this ticket is to implement a new form control and use it on the edit form. I've merged the latest master changes to have better visualization. Try to edit an existing widget group. Regarding Save/Cancel, I believe there was another ticket for adding this functionality.

image
image

File: src/app/modules/ix-forms/components/ix-icon-group/ix-icon-group.component.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<ix-label
+  *ngIf="label || tooltip"
+  [label]="label"
+  [tooltip]="tooltip"
+  [required]="required"
+></ix-label>
+
+<div class="icon-group" [attr.aria-label]="label">
+  @for (option of options; track option) {
```
Comment (AlexKarpov98) [inline src/app/modules/ix-forms/components/ix-icon-group/ix-icon-group.component.html:9]: for me adding tabbing is more readable.
```
  @for (option of options; track option) {
    <button
      mat-icon-button
      type="button"
      [disabled]="isDisabled"
      [ixTest]="[controlDirective.name, option[0]]"
      [attr.aria-label]="option[0]"
      [class.selected]="value === option[0]"
      (click)="onValueChanged(option[0])"
      (blur)="onTouch()"
    >
      <ix-icon [name]="option[1]"></ix-icon>
    </button>
  }
  @empty {
    <span>{{ 'No options are passed' | translate }}</span>
  }
```
But in docs they have not added tabbing as well.. https://angular.io/api/core/for

**Comment by AlexKarpov98** [comment]
@denysbutenko - make these items focusable, so we can change layout by TAB + ENTER.


https://github.com/truenas/webui/assets/22980553/0ad774da-1120-4de3-8525-c22ffc51ea7c

**Review by RehanY147 — CHANGES_REQUESTED** [review]
image
Save button is still missing. Should there be a save button on the form? A way to close the form successfully?

**Comment by RehanY147** [comment]
Please ping me when new changes/discussions are ready to be reviewed

File: src/app/modules/ix-forms/components/ix-icon-group/ix-icon-group.component.html (hunk 2 above)
Comment (denysbutenko) [reply src/app/modules/ix-forms/components/ix-icon-group/ix-icon-group.component.html:9]: This code follows existing linter rules.

**Comment by denysbutenko** [comment]
> image Save button is still missing. Should there be a save button on the form? A way to close the form successfully?

@RehanY147 No plans to add form buttons in this PR. The ticket concerns the layout selector, not the edit form.

**Comment by denysbutenko** [comment]
> @denysbutenko - make these layout items focusable, so we can change layout by TAB + ENTER.
> 
>  Screen.Recording.2024-04-26.at.12.07.14.mov

Good catch. The new syntax supports iterable collections (e.g., Map) but has an issue with DOM. It is important to remember to use `keyvalue` pipe to transform the Map into an array.

**Comment by AlexKarpov98** [comment]
@denysbutenko - the issue I mentioned is gone, next step and probably another ticket - to adjust rendered layout items below to correspond to the selected layout. 👍 
Screenshot 2024-04-29 at 10 17 34

**Comment by denysbutenko** [comment]
Form-related issues will addressed in another ticket [NAS-128557](https://ixsystems.atlassian.net/browse/NAS-128557)

**Review by AlexKarpov98 — APPROVED** [review]
Works good. 👍

**Pull request merged**